conjure-go-client/httpclient/client.go

```diff
@@ -154,26 +163,19 @@ func (c *clientImpl) doOnce(
 
 	// 2. create the transport and client
 	// shallow copy so we can overwrite the Transport with a wrapped one.
-	clientCopy := c.client
+	clientCopy := *c.client.CurrentHTTPClient()
 	transport := clientCopy.Transport // start with the concrete http.Transport from the client
 
-	middlewares := []Middleware{
-		// must precede the error decoders because they return a nil response and the metrics need the status code of
-		// the raw response.
-		c.metricsMiddleware,
-		// must precede the client error decoder
-		b.errorDecoderMiddleware,
-		// must precede the body middleware so it can read the response body
-		c.errorDecoderMiddleware,
-	}
+	// request decoder must precede the client decoder
+	// must precede the body middleware so it can read the response body
+	transport = wrapTransport(transport, b.errorDecoderMiddleware, c.errorDecoderMiddleware)
 	// must precede the body middleware so it can read the request body
-	middlewares = append(middlewares, c.middlewares...)
-	middlewares = append(middlewares, b.bodyMiddleware)
-	for _, middleware := range middlewares {
-		if middleware != nil {
-			transport = wrapTransport(transport, middleware)
-		}
-	}
+	transport = wrapTransport(transport, c.middlewares...)
+	// must wrap inner middlewares because it mutates the return values
+	transport = wrapTransport(transport, b.bodyMiddleware)
+	// must be the outermost middleware to recover panics in the rest of the request flow
+	transport = wrapTransport(transport, c.recoveryMiddleware)
```

Comment:
Looks like this also gets added in the client `Build` method. Should we remove it from there in favor of wiring it up here to ensure it's the last middleware?